public/main.js

```diff
@@ -25,63 +28,100 @@ deleteCompletedTodoBtn.onclick = function() {
     });
 };
 
+filterButtonAll.onclick = function() {
+    filterTodos("all");
+};
+filterButtonActive.onclick = function() {
+    filterTodos("active");
+};
+filterButtonComplete.onclick = function() {
+    filterTodos("complete");
+};
+
 function createTodo(title, callback) {
-    var createRequest = new XMLHttpRequest();
-    createRequest.open("POST", "/api/todo");
-    createRequest.setRequestHeader("Content-type", "application/json");
-    createRequest.send(JSON.stringify({
+    var dataToSend = JSON.stringify({
```

Comment:
These functions are also using a combination of the promise from fetch and a callbacks, which is incorrect. Instead, they need to return the promise, and the caller will use the promise to get the data or deal with the error, if necessary. An example of this is https://github.com/github/fetch#handling-http-error-statuses. It would probably be a good idea to take a deeper look into promises later, e.g : https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Promise